dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -202,6 +202,7 @@ void PushingToViewsBlockOutputStream::process(const Block & block, size_t view_n
     try
     {
         BlockInputStreamPtr in;
+        std::unique_ptr<InterpreterSelectQuery> select;
```

Comment:
std::optional is Ok.

Reply:
InterpreterSelectQuery is not copyable/movable, I can fix this, but not sure that it should be.

Reply:
But we don't copy or move it. Only construct.

Reply:
BTW, this modification cannot be understood without a comment. If we don't add a comment, next developer will revert it back because the code looks unmotivated.

Reply:
>But we don't copy or move it. Only construct.

Maybe I'm missing something, but AFAICS it need to be copied, since it is not created it in place

>this modification cannot be understood without a comment

Indeed (already have such thoughts), will add a comment

Reply:
`std::optional::emplace`

Reply:
thanks (sigh)